From 23cb111327f6c53785630f99f439d96d8b40c823 Mon Sep 17 00:00:00 2001
From: Jeegar Patel <jeegar.patel@xilinx.com>
Date: Wed, 26 Aug 2020 22:24:13 -0700
Subject: [PATCH 1537/1852] v4l: xilinx: multi-scaler: Fix warnings for
 INCOMPATIBLE_PARAM

commit 5cc86b30c7bcd0a65d2cd5bf6023e4a4756ba67e from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the warnings for the INCOMPATIBLE_PARAM type.

The actual warnings are:

-Event incompatible_param: argument of type "__u8 *" is incompatible
 with parameter of type "char *"

Signed-off-by: Jeegar Patel <jeegar.patel@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-multi-scaler.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-multi-scaler.c b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
index 1057bbbbb0b4..6c191dfb53f5 100644
--- a/drivers/media/platform/xilinx/xilinx-multi-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
@@ -1711,7 +1711,7 @@ static int enum_fmt(struct xm2m_msc_dev *xm2msc, struct v4l2_fmtdesc *f)
 
 	/* Format found */
 	fmt = &formats[i];
-	strlcpy(f->description, fmt->name,
+	strlcpy((char *)f->description, (char *)fmt->name,
 		sizeof(f->description));
 	f->pixelformat = fmt->fourcc;
 
@@ -1743,9 +1743,10 @@ static int xm2msc_enum_fmt_vid_out(struct file *file, void *fh,
 static int xm2msc_querycap(struct file *file, void *fh,
 			   struct v4l2_capability *cap)
 {
-	strncpy(cap->driver, XM2MSC_DRIVER_NAME, sizeof(cap->driver) - 1);
-	strncpy(cap->card, XM2MSC_DRIVER_NAME, sizeof(cap->card) - 1);
-	snprintf(cap->bus_info, sizeof(cap->bus_info),
+	strncpy((char *)cap->driver, XM2MSC_DRIVER_NAME,
+		sizeof(cap->driver) - 1);
+	strncpy((char *)cap->card, XM2MSC_DRIVER_NAME, sizeof(cap->card) - 1);
+	snprintf((char *)cap->bus_info, sizeof(cap->bus_info),
 		 "platform:%s", XM2MSC_DRIVER_NAME);
 	/*
 	 * This is only a mem-to-mem video device. The STREAMING
-- 
2.31.1

